shared/validate/validate.go

```diff
@@ -802,3 +802,33 @@ func ParseNetworkVLANRange(vlan string) (int, int, error) {
 
 	return vlanRangeStart, vlanRangeEnd - vlanRangeStart + 1, nil
 }
+
+// ValidHostname checks the string is valid DNS hostname.
+func ValidHostname(name string) error {
```

Comment:
The functions in this package take (mostly) the form `Is...()` so `IsHostname()` which avoids the stutter of `validate.Valid...()` on every call.